From d21664cce1db8debe2528f36b1fbd2b8af9c9401 Mon Sep 17 00:00:00 2001
From: serge-sans-paille <sguelton@redhat.com>
Date: Wed, 12 Feb 2020 08:57:19 +0100
Subject: [PATCH] Fix integration of pass plugins with llvm dylib

Call llvm_process_pass_plugin from clang when in standalone mode.

Differential Revision: https://reviews.llvm.org/D74464
---
 clang/CMakeLists.txt | 1 +
 llvm/CMakeLists.txt  | 3 +--
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: llvm-toolchain-10-10.0.0~+rc5/clang/CMakeLists.txt
===================================================================
--- llvm-toolchain-10-10.0.0~+rc5.orig/clang/CMakeLists.txt
+++ llvm-toolchain-10-10.0.0~+rc5/clang/CMakeLists.txt
@@ -864,6 +864,7 @@
 
 if(CLANG_BUILT_STANDALONE)
   llvm_distribution_add_targets()
+  process_llvm_pass_plugins()
 endif()
 
 configure_file(
Index: llvm-toolchain-10-10.0.0~+rc5/llvm/CMakeLists.txt
===================================================================
--- llvm-toolchain-10-10.0.0~+rc5.orig/llvm/CMakeLists.txt
+++ llvm-toolchain-10-10.0.0~+rc5/llvm/CMakeLists.txt
@@ -1067,6 +1067,7 @@
 # after all targets are created.
 include(LLVMDistributionSupport)
 llvm_distribution_add_targets()
+process_llvm_pass_plugins()
 
 # This allows us to deploy the Universal CRT DLLs by passing -DCMAKE_INSTALL_UCRT_LIBRARIES=ON to CMake
 if (MSVC AND CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows" AND CMAKE_INSTALL_UCRT_LIBRARIES)
@@ -1091,5 +1092,3 @@
 if (LLVM_INCLUDE_UTILS AND LLVM_INCLUDE_TOOLS)
   add_subdirectory(utils/llvm-locstats)
 endif()
-
-process_llvm_pass_plugins()
